core/capabilities/remote/target/receiverRequest.go

```diff
@@ -0,0 +1,186 @@
+package target
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	commoncap "github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/pb"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/types"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+type remoteTargetCapabilityRequest struct {
+	lggr logger.Logger
+
+	capability capabilities.TargetCapability
+
+	capabilityPeerId p2ptypes.PeerID
+	capabilityID     string
+	capabilityDonID  string
+
+	dispatcher types.Dispatcher
+
+	requesters              map[p2ptypes.PeerID]bool
+	responseSentToRequester map[p2ptypes.PeerID]bool
+
+	createdTime time.Time
+
+	response      []byte
+	responseError types.Error
+
+	callingDon       commoncap.DON
+	requestMessageID string
+
+	requestTimeout time.Duration
+}
+
+func newTargetCapabilityRequest(lggr logger.Logger, capability capabilities.TargetCapability, capabilityID string, capabilityDonID string, capabilityPeerId p2ptypes.PeerID,
+	callingDon commoncap.DON, requestMessageID string,
+	dispatcher types.Dispatcher, requestTimeout time.Duration) *remoteTargetCapabilityRequest {
+	return &remoteTargetCapabilityRequest{
+		lggr:                    lggr,
+		capability:              capability,
+		createdTime:             time.Now(),
+		capabilityID:            capabilityID,
+		capabilityDonID:         capabilityDonID,
+		capabilityPeerId:        capabilityPeerId,
+		dispatcher:              dispatcher,
+		requesters:              map[p2ptypes.PeerID]bool{},
+		responseSentToRequester: map[p2ptypes.PeerID]bool{},
+		callingDon:              callingDon,
+		requestMessageID:        requestMessageID,
+		requestTimeout:          requestTimeout,
+	}
+}
+
+func (e *remoteTargetCapabilityRequest) receive(ctx context.Context, msg *types.MessageBody) error {
```

Comment:
Did you consider using MessageCache for this logic? It would be nice to implement similar behaviors in a consistent way across all remote capabilities.